src/core/system_parameters/Beidou_B1C.h

```diff
@@ -0,0 +1,242 @@
+/*!
+ * \file Beidou_B1C.h
+ * \brief Defines system parameters for Beidou B1C signal
+ * \author Andrew Kamble, 2019. andrewkamble88@gmail.com 
+ * \note Code added as part of GSoC 2019 program
+ * Detailed description of the file here if needed.
+ *
+ * -------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2019  (see AUTHORS file for a list of contributors)
+ *
+ * GNSS-SDR is a software defined Global Navigation
+ *          Satellite Systems receiver
+ *
+ * This file is part of GNSS-SDR.
+ *
+ * GNSS-SDR is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * GNSS-SDR is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNSS-SDR. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#ifndef GNSS_SDR_BEIDOU_B1C_H_
+#define GNSS_SDR_BEIDOU_B1C_H_
+
+
+#include "MATH_CONSTANTS.h"
+#include "gnss_frequencies.h"
+#include <cstdint>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+const double   BEIDOU_B1C_FREQ_HZ = FREQ1;              //!< BEIDOU B1C carrier frequency [Hz]
+const uint32_t BEIDOU_B1C_OPT_ACQ_FS_HZ = 2000000;  //!< Sampling frequncy that maximizes the acquisition SNR while using a non-multiple of chip rate
+const int32_t  BEIDOU_B1C_SAMPLES_PER_SYMBOL = 1;
+
+const double  BEIDOU_B1Cd_CODE_RATE_HZ = 1.023e6;      //!< BEIDOU_B1C data code rate [chips/s]
+const int32_t BEIDOU_B1Cd_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1C data  code length [chips]
+const double  BEIDOU_B1Cd_PERIOD = 0.01;               //!< BEIDOU_B1C data code period [seconds]
+const int32_t BEIDOU_B1Cd_CODE_PERIOD_MS = 10;
+
+const double  BEIDOU_B1Cp_CODE_RATE_HZ = 1.023e6;      //!< BEIDOU_B1c pilot code rate [chips/s]
+const int32_t BEIDOU_B1Cp_CODE_LENGTH_CHIPS = 10230;  //!< BEIDOU_B1c pilot code length [chips]
+const double  BEIDOU_B1Cp_PERIOD = 0.01;               //!< BEIDOU_B1C data code period [seconds]
+const int32_t BEIDOU_B1Cp_PERIOD_MS = 10;
+
+const int32_t BEIDOU_NBR_SATS = 63;		      //!<Total number of satellites
```

Comment:
This label is very generic, consider renaming to something like BEIDOU_B1C_NBR_SATS to avoid issues when integrating the code